### PR TITLE
List.iter already checks for []

### PR DESCRIPTION
**Pull request by forki**

### CODE REVIEW

File: src/fsharp/ErrorLogger.fs [hunk 1]
```diff
@@ -518,10 +518,7 @@ type OperationResult<'T> =
     
 type ImperativeOperationResult = OperationResult<unit>
 
-let ReportWarnings warns = 
-    match warns with 
-    | [] -> () // shortcut in common case
-    | _ -> List.iter warning warns
+let inline ReportWarnings warns = List.iter warning warns
```
Comment (KevinRansom) [inline src/fsharp/ErrorLogger.fs]: the [] check is an optimization that cuts out a couple of stack frames, before the check can occur.  Looking at this, it doesn't seem to occur often enough to warrant either the optimization or the in-line.  Can we lose the inline too?

**Comment by forki** [comment]
I personally feel that the inline makes sense here as it's just giving
domain specific name to iter.

Am 06.03.2018 01:36 schrieb "Kevin Ransom (msft)" <notifications@github.com

**Comment by KevinRansom** [comment]
@forki ... I may be wrong but in my head, inline is an indicator that performance is a goal.  Or at least I would prefer that it is used as an indicator that it is on a hot path or some such.

The source code at the call site is unchanged regardless of the use of inline ... and so the domain specific name is still valid.

Anyway my 2 cents.

**Comment by forki** [comment]
I personally see it as zero-cost abstraction. But tbh this thing is probably not worth discussing it too long. just close it if you think it's not worth it.

**Comment by KevinRansom** [comment]
It's not an abstraction.  It may change code gen, but it changes nothing about the programmer view of the code.  It is also not zero cost, it makes inline a less significant indicator of performance sensitivity.

**Comment by dsyme** [comment]
@forki Agreed with @KevinRansom here

**Pull request closed without merging**